include/sensors/core/fdc1004.hpp

```diff
@@ -27,39 +27,467 @@
 namespace sensors {
 namespace fdc1004 {
 
-// Capacitance Sensor Address and Registers
 constexpr uint16_t ADDRESS = 0x50 << 1;
-constexpr uint8_t MSB_MEASUREMENT_1 = 0x00;
-constexpr uint8_t LSB_MEASUREMENT_1 = 0x01;
-constexpr uint8_t CONFIGURATION_MEASUREMENT = 0x08;
-constexpr uint8_t FDC_CONFIGURATION = 0x0C;
-constexpr uint8_t DEVICE_ID_REGISTER = 0xFF;
-
-// CHA
-constexpr uint16_t POSITIVE_INPUT_CHANNEL = 0x0;
-// CHB
-constexpr uint16_t NEGATIVE_INPUT_CHANNEL = 0x4 << 10;
-// configurations: our reads will be differential reads of
-// the U.FL connector on one channel and the internal
-// common-mode compensator, CAPDAC, on the other.
-constexpr uint16_t DEVICE_CONFIGURATION =
-    0x0 << 13 |  // CHA = CIN1 (U.FL Connector)
-    0x4 << 10;   // CHB = CAPDAC
-constexpr uint8_t DEVICE_CONFIGURATION_MSB =
-    static_cast<uint8_t>(DEVICE_CONFIGURATION >> 8);
-constexpr uint8_t DEVICE_CONFIGURATION_LSB =
-    static_cast<uint8_t>(DEVICE_CONFIGURATION & 0xff);
-constexpr uint16_t SAMPLE_RATE = 1 << 10 |  // 100S/s
-                                 1 << 8 |   // Repeat enabled
-                                 1 << 7;    // Measurement 1 enabled
-
-constexpr uint8_t SAMPLE_RATE_MSB = static_cast<uint8_t>(SAMPLE_RATE >> 8);
-constexpr uint8_t SAMPLE_RATE_LSB = static_cast<uint8_t>(SAMPLE_RATE & 0xff);
-constexpr uint16_t DEVICE_ID = 0x1004;
 
-constexpr uint8_t CAPDAC_MSB_MASK = 0x3;
+enum class CHA : uint8_t {
+    CIN1 = 0x0,
+    CIN2 = 0x1,
+    CIN3 = 0x2,
+    CIN4 = 0x3,
+};
+
+enum class CHB : uint8_t {
+    CIN1 = 0x0,
+    CIN2 = 0x1,
+    CIN3 = 0x2,
+    CIN4 = 0x3,
+    CAPDAC = 0x4,
+    DISABLED = 0x8
+};
+
+enum class MeasurementRate : uint8_t {
+    RESERVED = 0x0,
+    ONE_HUNDRED_SAMPLES_PER_SECOND = 0x1,
+    TWO_HUNDRED_SAMPLES_PER_SECOND = 0x2,
+    FOUR_HUNDRED_SAMPLES_PER_SECOND = 0x3,
+};
+
+enum class MeasureConfigMode : uint8_t { ONE, TWO, THREE, FOUR };
```

Comment:
if this goes to the device we should define enumeration values explicitly

Reply:
It does not. It's only used in the `driver` portion of the code.